shared-bindings/tilepalettemapper/TilePaletteMapper.c

```diff
@@ -0,0 +1,165 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Tim Cocks for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+#include <stdint.h>
+#include <stdlib.h>
+#include "shared/runtime/context_manager_helpers.h"
+#include "py/binary.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "shared-bindings/util.h"
+#include "shared-bindings/displayio/Palette.h"
+#include "shared-bindings/tilepalettemapper/TilePaletteMapper.h"
+
+//| class TilePaletteMapper:
+//|     """Contains a displayio.Palette a series of mappings from one palette color_index to another palette color_index.
+//|     One mapping for tile in a 2D grid. Can be set as the pixel_shader for a TileGrid to re-map the colors for
+//|     specific tile(s) within the TileGrid."""
+//|
+//|     def __init__(self, palette: displayio.Palette, width: int, height: int) -> None:
```

Comment:
I think you'll want the number of colors used in the source bitmap too. That way you can validate the length/index of colors set. Bitmap has a `bits_per_value` property that could give you an upper bound for this. Or the palette returned by image load would give you the number in the actual image too.

Reply:
updated with latest commit to accept `input_color_count`